lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+   * @param {boolean} verboseErrors
+   * @returns {object}
+   * @throws InternalError
+   */
+  curateFieldSpecificationFormat(fieldSpec, indexName, collectionName, fieldName, verboseErrors) {
+    var errors = [];
+
+    verboseErrors = verboseErrors || false;
+
+    if (!checkAllowedProperties(fieldSpec, ['mandatory', 'type', 'defaultValue', 'description', 'multivalued', 'typeOptions'])) {
+      throwOrStoreError(`The field ${indexName}.${collectionName}.${fieldName} specification has invalid properties.`, verboseErrors, errors);
+    }
 
-  if (!checkAllowedProperties(fieldSpec, ['mandatory', 'type', 'defaultValue', 'description', 'multivalued', 'typeOptions'])) {
-    throwOrStoreError(`The field ${indexName}.${collectionName}.${fieldName} specification has invalid properties.`, verboseErrors, errors);
-  }
+    mandatoryFieldSpecProperties.forEach(propertyName => {
+      if (!fieldSpec.hasOwnProperty(propertyName)) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, ${propertyName} is a mandatory field specification property.`, verboseErrors, errors);
+      }
+    });
 
-  mandatoryFieldSpecProperties.forEach(propertyName => {
-    if (!fieldSpec.hasOwnProperty(propertyName)) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, ${propertyName} is a mandatory field specification property.`, verboseErrors, errors);
+    if (!this.types[fieldSpec.type]) {
+      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}: ${fieldSpec.type} is not a recognized type.`, verboseErrors, errors);
     }
-  });
 
-  if (!this.types[fieldSpec.type]) {
-    throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}: ${fieldSpec.type} is not a recognized type.`, verboseErrors, errors);
-  }
+    if (fieldSpec.hasOwnProperty('multivalued')) {
+      if (!checkAllowedProperties(fieldSpec.multivalued, ['value', 'minCount', 'maxCount'])) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, the multivalued field specification has invalid properties.`, verboseErrors, errors);
+      }
 
-  if (fieldSpec.hasOwnProperty('multivalued')) {
-    if (!checkAllowedProperties(fieldSpec.multivalued, ['value', 'minCount', 'maxCount'])) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, the multivalued field specification has invalid properties.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.hasOwnProperty('value')) {
+        throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "value" is a mandatory property for multivalued field specification.`, verboseErrors, errors);
+      }
 
-    if (!fieldSpec.multivalued.hasOwnProperty('value')) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "value" is a mandatory property for multivalued field specification.`, verboseErrors, errors);
-    }
+      if (!fieldSpec.multivalued.value && (fieldSpec.multivalued.hasOwnProperty('minCount'))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 95 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)